### ISSUE TITLE
ipfs-cluster-service 0.12.1 unhappy with CLUSTER_SERVICE variable that 0.10.x does not complain 

### ISSUE THREAD

**Issue by stephgosling** [ipfs/ipfs-cluster#1005]
TL;DR: `error applying environment variables to configuration: error loading cluster secret from config: encoding/hex: invalid byte: U+0068 'h'` Explanation follows

**Additional information:**
 - OS: Ubuntu 18.04 x64
 - IPFS Cluster version: 0.12.1
 - Installation method:  https://dist.ipfs.io/ipfs-cluster-service/v0.12.1/ipfs-cluster-service_v0.12.1_linux-amd64.tar.gz

Hello! (I <3 ipfs-cluster).

I'm trying to stand up a new cluster (0.12 as above) to supercede an 0.10 one I have. In the 0.10 cluster I was managing raft by passing `CLUSTER_SECRET` as an envvar in the shell. In 0.12.1 if I set that variable ipfs-cluster-service spits encoding errors at me:

```
ipfs@ipfs-dev-001-0ad6c94322833b45e:~$ env | grep CLUSTER
CLUSTER_SECRET=hello
ipfs@ipfs-dev-001-0ad6c94322833b45e:~$ ipfs-cluster-service init
error applying environment variables to configuration: error loading cluster secret from config: encoding/hex: invalid byte: U+0068 'h'
```
so unset:
```
ipfs@ipfs-dev-001-0ad6c94322833b45e:~$ unset CLUSTER_SECRET
ipfs@ipfs-dev-001-0ad6c94322833b45e:~$ ipfs-cluster-service init
23:49:28.623  INFO     config: Saving configuration config.go:479
configuration written to /home/ipfs/.ipfs-cluster/service.json.
23:49:28.624  INFO     config: Saving identity identity.go:73
new identity written to /home/ipfs/.ipfs-cluster/identity.json
new empty peerstore written to /home/ipfs/.ipfs-cluster/peerstore.
ipfs@ipfs-dev-001-0ad6c94322833b45e:~$ 
```
however interesting:

```
ipfs@ipfs-dev-001-0ad6c94322833b45e:~$ export CLUSTER_SECRET=foo
ipfs@ipfs-dev-001-0ad6c94322833b45e:~$ ipfs-cluster-service daemon
23:50:08.373  INFO    service: Initializing. For verbose output run with "-l debug". Please wait... daemon.go:46
error loading configurations: error loading cluster secret from config: encoding/hex: invalid byte: U+006F 'o'
```

I am going to continue to prod this but for me, if `CLUSTER_SECRET` is set, my new daemon won't start, and that's a regression.

**Comment by stephgosling** [comment]
ok this appears to be environmental as _sometimes_ I can get it to init properly

**Comment by stephgosling** [comment]
Answering my own question, CLUSTER_SECRET needs to be hex encoded (as per the docs, this bug is invalid)

**Comment by hsanjuan** [comment]
> Answering my own question, CLUSTER_SECRET needs to be hex encoded (as per the docs, this bug is invalid)

Glad you solved it!
